xbmc/cores/VideoPlayer/Process/VideoBuffer.h

```diff
@@ -99,7 +99,7 @@ class CVideoBuffer
   virtual uint8_t* GetMemPtr() { return nullptr; };
   virtual void GetPlanes(uint8_t*(&planes)[YuvImage::MAX_PLANES]) {};
   virtual void GetStrides(int(&strides)[YuvImage::MAX_PLANES]) {};
-  virtual void SetDimensions(int width, int height, const int (&strides)[YuvImage::MAX_PLANES]) {};
+  virtual void SetDimensions(int width, int height, const int (&strides)[YuvImage::MAX_PLANES], const int planeOffsets[YuvImage::MAX_PLANES]) {};
```

Comment:
> but this would lead to not small code duplication in CVideoBufferSysMem

does not require any kind of code dupe. just call the 3-arguments method and set data pointer with offsets.